pygeos/lib/geom.pyx

```diff
@@ -0,0 +1,90 @@
+from cpython cimport PyObject
+cimport cython
+
+import numpy as np
+cimport numpy as np
+
+from pygeos.lib.geos_wrapper cimport (
+    GEOSContextHandle_t,
+    GEOSGeometry,
+    GEOSGeom_clone_r,
+    GEOSGetGeometryN_r,
+    GEOSGetNumGeometries_r,
+    GEOS_init_r
+)
+from pygeos.lib.pygeos_wrapper cimport (
+    import_pygeos_core_api,
+    PyGEOS_CreateGeometry,
+    PyGEOS_GetGEOSGeometry
+)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef geos_get_num_geometries(object[:] array):
```

Comment:
Indeed, we do not need it redeclared.  Originally I thought there would be performance gain in avoiding the ufunc, but according to my benchmarks, both versions are roughly equivalent.

Reply:
>  Originally I thought there would be performance gain in avoiding the ufunc, but according to my benchmarks, both versions are roughly equivalent.

I think generally speaking, the ufunc function or a cython written loop have more or less the same speed (at least when we started with pygeos, I compared a few things with the older cython branch in geopandas, and that was the conclusion. And I think this is also as expected, since both ensure the looping is done in C, so reducing the python overhead, but then the remaining bulk is still the GEOS function itself, which is the same for both). 

But so in practice in cython code here, the only difference is then that calling the ufunc is actually calling a python function (with an overhead of calling into python from cython), while the redefined cython function can be called from the other cython function without python overhead. 
But as long as it's only called once in the "setup" part of the other function, so not within the loop, that overhead is not actually significant. Other functions we call before the loop like `np.arange` or `np.empty` already have the same "issue" as well, those also cause python interaction. But as long as this python interaction doesn't happen in the loop, that's fine.